content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md

```diff
@@ -0,0 +1,420 @@
+	subject_pattern = "organization:<organization-name>:pipeline:<pipeline-name>:ref:refs/heads/main:commit:[0-9a-f]+:step:"
+  }
+}
+
+output "buildkite-identity" {
+  value = chainguard_identity.buildkite.id
+}
+
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.buildkite.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+EOF
+```
+
+Following that, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+
+## Creating Your Resources
+
+First, run `terraform init` to initialize Terraform's working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you're satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 3 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + buildkite-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `buildkite-identity` value.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+buildkite-identity = "<your buildkite identity>"
+```
+
+This is the identity's UIDP (unique identity path), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
+
+```sh
+chainctl iam identities ls
+```
+
+You're now ready to edit a Buildkite pipeline in order to test out this identity.
+
+
+## Testing the identity with a Buildkite pipeline
+
+To test the identity you created with Terraform in the previous section, navigate to your Buildkite pipeline. From the Buildkite Dashboard, click **Pipelines** in the top navigation bar and then click on the pipeline you specified in the `buildkite.tf` file. 
+
+From there, click the **Edit Steps** button to add the following commands to a step in your pipeline.
+
+```
+- command: |
+	wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"
+	chmod +x chainctl
+	mv chainctl /usr/bin
```

Comment:
```suggestion
```

I don't think it needs to get mv'ed anywhere - better to keep it confined to the build directory and then run `./chainctl auth ...` in the next step.